src/storage/test/QueryBoundBenchmark.cpp

```diff
@@ -0,0 +1,192 @@
+            TestUtils::propDef(cpp2::PropOwner::SOURCE,
+                               folly::stringPrintf("tag_%d_col_%d", 3001 + i*2, i*2),
+                               3001 + i*2));
+    }
+    tmpColumns.emplace_back(TestUtils::propDef(cpp2::PropOwner::EDGE,
+                                               folly::stringPrintf("_dst")));
+    tmpColumns.emplace_back(TestUtils::propDef(cpp2::PropOwner::EDGE,
+                                               folly::stringPrintf("_rank")));
+    // Return edge props col_0, col_2, col_4 ... col_18
+    for (int i = 0; i < 10; i++) {
+        tmpColumns.emplace_back(
+            TestUtils::propDef(cpp2::PropOwner::EDGE,
+                               folly::stringPrintf("col_%d", i*2)));
+    }
+    req.set_return_columns(std::move(tmpColumns));
+    return req;
+}
+
+}  // namespace storage
+}  // namespace nebula
+
+void run(int32_t iters, int32_t handlerNum) {
+    FLAGS_max_handlers_per_req = handlerNum;
+    nebula::storage::cpp2::GetNeighborsRequest req;
+    BENCHMARK_SUSPEND {
+        req = nebula::storage::buildRequest();
+    }
+    auto executor = std::make_unique<folly::CPUThreadPoolExecutor>(FLAGS_handler_num);
+    for (decltype(iters) i = 0; i < iters; i++) {
+        auto* processor
+                = nebula::storage::QueryBoundProcessor::instance(
+                                                            gKV.get(),
+                                                            schema.get(),
+                                                            executor.get(),
+                                                            nebula::storage::BoundType::OUT_BOUND);
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+    }
+}
+
+BENCHMARK(query_bound_1, iters) {
+    run(iters, 1);
+}
+
+BENCHMARK(query_bound_3, iters) {
+    run(iters, 3);
+}
+
+BENCHMARK(query_bound_10, iters) {
+    run(iters, 10);
+}
+/*************************
+ * End of benchmarks
+ ************************/
+
+
+int main(int argc, char** argv) {
+    folly::init(&argc, &argv, true);
+    nebula::fs::TempDir rootPath("/tmp/QueryBoundBenchmarkTest.XXXXXX");
+    nebula::storage::setUp(rootPath.path());
+    folly::runBenchmarks();
+    gKV.reset();
+    return 0;
+}
+/*
+40 processors, Intel(R) Xeon(R) CPU E5-2690 v2 @ 3.00GHz.
+req_parts=1, vrpp=100, handler_num=10
+Notes: all requests hit the cache.
+
+============================================================================
+QueryBoundBenchmark.cpprelative                           time/iter  iters/s
+============================================================================
+query_bound_1                                              122.30ms     8.18
+query_bound_3                                               41.18ms    24.29
+query_bound_10                                              13.69ms    73.04
+============================================================================
```

Comment:
Yes, the performance is linear under light load.  Under heavy load,  to avoid a big request hold all processors,  i have set the max handlers used limit with one flag.

Reply:
We don't have enough stress tests. maybe a TODO to remember this.

Reply:
Fair enough!